From c2b3a46dcf6518bb087ea3650ee04d05e519e46f Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Thu, 31 May 2018 14:11:34 +0200
Subject: [PATCH 11/14] SimSettings: Fix preferred calls sim not being disabled

* When disabling a sim on a dual sim device, the selection
  would not be disabled properly

Change-Id: Ib8054f724ae974640491d03c668066db47b8a3a9
---
 src/com/android/settings/sim/SimSettings.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/settings/sim/SimSettings.java b/src/com/android/settings/sim/SimSettings.java
index 21dea200dc..8bd120ac5e 100644
--- a/src/com/android/settings/sim/SimSettings.java
+++ b/src/com/android/settings/sim/SimSettings.java
@@ -247,7 +247,7 @@ public class SimSettings extends RestrictedSettingsFragment implements Indexable
         simPref.setSummary(account == null
                 ? mContext.getResources().getString(R.string.sim_calls_ask_first_prefs_title)
                 : (String) account.getLabel());
-        simPref.setEnabled(allPhoneAccounts.size() > 1);
+        simPref.setEnabled(allPhoneAccounts.size() > 1 && mSelectableSubInfos.size() > 1);
     }
 
     @Override
-- 
2.17.1

